src/systems/triggered_publisher/TriggeredPublisher.cc

```diff
@@ -645,6 +703,71 @@ void TriggeredPublisher::Configure(const Entity &,
       std::thread(std::bind(&TriggeredPublisher::DoWork, this));
 }
 
+//////////////////////////////////////////////////
+void TriggeredPublisher::PublishMsg(std::size_t pending)
+{
+    for (auto &info : this->outputInfo)
+    {
+      for (std::size_t i = 0; i < pending; ++i)
+      {
+        info.pub.Publish(*info.msgData);
+      }
+    }
+}
+
+//////////////////////////////////////////////////
+void TriggeredPublisher::CallService()
+{
+  for (auto &serviceInfo : this->servOutputInfo)
+  {
+    bool result;
+    auto req = msgs::Factory::New(serviceInfo.reqType,
+                                  serviceInfo.reqMsg);
+    if (!req)
+    {
+      ignerr << "Unable to create request for type ["
+             << serviceInfo.reqType << "].\n";
+      return;
+    }
+
+    auto rep = msgs::Factory::New(serviceInfo.repType);
+    if (!rep)
+    {
+      ignerr << "Unable to create response for type ["
+             << serviceInfo.repType << "].\n";
+      return;
+    }
+
+    bool executed = this->node.Request(serviceInfo.servName,
+                                       *req,
+                                       serviceInfo.timeout,
+                                       *rep,
+                                       result);
```

Comment:
This can fit in two lines. We don't need to put one arg per line here - that is only done when it gets hard to read, or if it's a long list and we expect to modify them often in the future, then one line each makes the diffs more clear.